File: 2022/05/22/fizzBuzz.py
"""
https://leetcode.com/problems/fizz-buzz/
Author: Andrew Jarombek
Date: 5/22/2022
"""

from typing import List


def fizz_buzz(n: int) -> List[str]:
    result = []

    for i in range(1, n + 1):
        val = ""

        if i % 3 == 0:
            val += "Fizz"

        if i % 5 == 0:
            val += "Buzz"

        if len(val) == 0:
            val = f"{i}"

        result.append(val)

    return result

File: 2023/06/19/go/fizz_buzz.go
/**
 * https://leetcode.com/problems/fizz-buzz/
 * Author: Andrew Jarombek
 * Date: 6/19/2023
 */

package _go

import "strconv"

func fizzBuzz(n int) []string {
	result := make([]string, n)

	for i := 1; i <= n; i++ {
		if i%3 == 0 && i%5 == 0 {
			result[i-1] = "FizzBuzz"
		} else if i%3 == 0 {
			result[i-1] = "Fizz"
		} else if i%5 == 0 {
			result[i-1] = "Buzz"
		} else {
			result[i-1] = strconv.Itoa(i)
		}
	}

	return result
}
